test/test_helper.rb

```diff
@@ -2,13 +2,15 @@
 
 ENV['RAILS_ENV'] ||= 'test'
 
+require 'minitest/autorun'
+
 # simplecov and codacy coverage
 require 'simplecov'
-require 'codacy-coverage'
+require 'codacy-coverage' if ENV['CI']
```

Comment:
Buraya dikkat, bu güzel bir trick. Circleci, codacy, codebeat vb. araçlar ortak olarak `ENV['CI']`'ı tanıyor. Sadece development ve test ortamında çalışmasını istediğiniz, circle'da çalışmasını istemediğiniz durumlarda çok kullanışlı. Bu durumlara bir örnek KPS sorguları. KPS'ye sorgu yapabilmek için üniversitenin izin verdiği bir IP adresi gerekiyor. Dolayısıyla bu testten vazgeçmeden, sadece bir testin CI ortamında çalışmamasını sağlamak için iki seçenek var.

1) CircleCI'ın tüm IP bloğuna izin tanımlamak - ki tek bir IP kullanmadığı ve AWS'de deploy edildiği için aslında tüm AWS IP bloğuna servisleri açmak demek - sakıncalı.
2) `ENV['CI']` ile tek bir testi CI'da pas geçmek.

Reply:
Burada ise codacy coverage'a sadece CI ortamında raporlama yapmak için kullandım. Yerelde zaten codecov'un raporlarına bakıyoruz ve yerelde istesek bile codacy'ye raporlamıyor. Dökümanlarda net verilmemiş bir bilgi, şurdan yakaladım:

https://github.com/codacy/ruby-codacy-coverage/blob/master/lib/codacy/formatter.rb#L19